packages/sdk-communication-layer/src/services/SocketService/ConnectionManager/connectToChannel.ts

```diff
@@ -15,14 +17,14 @@ import { setupChannelListeners } from '../ChannelManager';
  * @param instance The current instance of the SocketService.
  * @throws {Error} Throws an error if the socket is already connected.
  */
-export function connectToChannel({
+export async function connectToChannel({
   options,
   instance,
 }: {
   options: ConnectToChannelOptions;
   instance: SocketService;
-}) {
-  const { channelId, withKeyExchange } = options;
+}): Promise<void> {
+  const { channelId, authorized, withKeyExchange } = options;
```

Comment:
maybe it would be good to do some destructuring of `instance` as well as it's used so many times below